sharding/test/compression_verifier_test.go

```diff
@@ -0,0 +1,72 @@
+package test
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/Shopify/ghostferry"
+	sth "github.com/Shopify/ghostferry/sharding/testhelpers"
+	"github.com/Shopify/ghostferry/testhelpers"
+	"github.com/stretchr/testify/suite"
+)
+
+const ()
+
+type CompressionVerifierTestSuite struct {
+	*sth.ShardingUnitTestSuite
+}
+
+func (t *CompressionVerifierTestSuite) SetupTest() {
+	t.ShardingUnitTestSuite.SetupTest()
+}
+
+func (t *CompressionVerifierTestSuite) TearDownTest() {
+	t.ShardingUnitTestSuite.TearDownTest()
+}
+
+func (t *CompressionVerifierTestSuite) TestFailsVerificationForDifferentCompressedData() {
+	tableCompressions := make(ghostferry.TableColumnCompressionConfig)
+	tableCompressions[testhelpers.TestCompressedTable1Name] = make(map[string]string)
+	tableCompressions[testhelpers.TestCompressedTable1Name][testhelpers.TestCompressedColumn1Name] = ghostferry.CompressionSnappy
+	t.Config.TableColumnCompression = tableCompressions
+
+	t.Require().NotEqual(testhelpers.TestCompressedData1, testhelpers.TestCompressedData2)
+	t.InsertCompressedRowInDb(43, "gftest1", testhelpers.TestCompressedData1, t.Ferry.Ferry.SourceDB)
+	t.InsertCompressedRowInDb(43, "gftest2", testhelpers.TestCompressedData2, t.Ferry.Ferry.TargetDB)
+
+	err := t.Ferry.Start()
+	t.Require().Nil(err)
+
+	errHandler := &testhelpers.ErrorHandler{}
+	t.Ferry.Ferry.ErrorHandler = errHandler
+
+	t.Ferry.Run()
+
+	t.Require().NotNil(errHandler.LastError)
+	t.Require().Equal("verifier detected data discrepancy: verification failed on table: gftest1.test_compressed_table_1 for pks: 43", errHandler.LastError.Error())
+}
+func (t *CompressionVerifierTestSuite) TestCanCopyDifferentCompressedDataButEqualDecompressedData() {
+	tableCompressions := make(ghostferry.TableColumnCompressionConfig)
+	tableCompressions[testhelpers.TestCompressedTable1Name] = make(map[string]string)
+	tableCompressions[testhelpers.TestCompressedTable1Name][testhelpers.TestCompressedColumn1Name] = ghostferry.CompressionSnappy
+	t.Config.TableColumnCompression = tableCompressions
+
+	t.Require().NotEqual(testhelpers.TestCompressedData3, testhelpers.TestCompressedData4)
+
+	t.InsertCompressedRowInDb(43, "gftest1", testhelpers.TestCompressedData3, t.Ferry.Ferry.SourceDB)
+	t.InsertCompressedRowInDb(43, "gftest2", testhelpers.TestCompressedData4, t.Ferry.Ferry.TargetDB)
+
+	err := t.Ferry.Start()
+	t.Require().Nil(err)
+
+	t.Ferry.Run()
```

Comment:
Should we be instantiating and checking an `ErrorHandler` in addition to checking the error returned from `t.Ferry.Start()` like the test above to ensure no errors are returned during verification?

Reply:
The default ErrorHandler panics if there's an error in the Ferry, not the greatest but it works and matches production